packages/dd-trace/src/plugins/util/web.js

```diff
@@ -337,6 +351,59 @@ const web = {
 
     return `${path}?${qs}`
   },
+
+  extractIp (context) {
+    const { req, config } = context
+
+    if (config.clientIpHeaderDisabled) return null
+
+    const headers = req.headers
+    let foundIp
+
+    if (config.clientIpHeader) {
+      const header = headers[config.clientIpHeader]
+      if (!header) return null
+
+      foundIp = findFirstIp(header)
+    } else {
+      const foundHeaders = []
+
+      for (let i = 0; i < ipHeaderList.length; i++) {
+        if (headers[ipHeaderList[i]]) {
+          foundHeaders.push(ipHeaderList[i])
+        }
+      }
+
+      if (foundHeaders.length === 0) {
```

Comment:
This whole block could be simplified by always looping and setting all tags. This is also more in line with future plans of never altering the number of tags.

Reply:
But we don't know which tags are to be sent at that point in time, that's the whole problem. We'll have to fix that "never altering the number of tags" once we have the architecture for it i'd say.

Reply:
Just always set the tags. If it's empty, for example because of joining an empty list, it will be automatically ignored by the exporter.